community/cabal/APKBUILD

```diff
@@ -2,61 +2,56 @@
 # Maintainer: Mitch Tishmack <mitch.tishmack@gmail.com>
 pkgname=cabal
 pkgver=1.24.0.2
-pkgrel=0
+pkgrel=1
 pkgdesc="The Haskell Cabal"
 url="http://haskell.org"
 arch="armhf x86_64"
 license="BSD3"
 depends="gmp zlib"
-# Note, this is a temporary state of affairs using the bootstrap.sh script.
-#
-# Once apkbuilds for ghc pkgs are working the pkg sources can be moved into
-# the makedepends lines.
 makedepends="ghc ghc-dev gmp-dev libffi-dev zlib-dev binutils-gold chrpath"
 subpackages="$pkgname-doc"
 source="https://hackage.haskell.org/package/$pkgname-install-$pkgver/$pkgname-install-$pkgver.tar.gz
-	cabal-0001-force-ld.gold.patch
-	cabal-0002-busybox-od.patch
-	cabal-0003-use-apkbuild-downloads.patch
-
 	https://hackage.haskell.org/package/mtl-2.2.1/mtl-2.2.1.tar.gz
-	https://hackage.haskell.org/package/mtl-2.2.1/mtl.cabal
+	mtl.cabal::https://hackage.haskell.org/package/mtl-2.2.1/revision/1.cabal
```

Comment:
If the requirement is that this file has to be named mtl-2.2.1.cabal, then this won't ever work. The cabal file here is updated outside of the package-version.tar.gz to allow for newer ghc compilers to be able to bootstrap cabal itself.

As such, this is how it has to be, otherwise we cannot bootstrap cabal without cabal already being installed.